### PR TITLE
BOLT 2: introduce alt_addr feature for private reconnections

### PR DESCRIPTION
**Pull request by maxrantil**
This PR adds documentation for a new feature: the `alt_addr` message in Bolt 02. The `alt_addr` message allows nodes to specify a private alternative connection address for peers, improving privacy, reliability, and latency in established connections.

### CODE REVIEW

**Comment by t-bast** [comment]
I'm not sure that this is generally useful, and thus probably doesn't meet the bar for inclusion in the BOLTs. Maybe a [bLIP](https://github.com/lightning/blips) would make more sense?

**Comment by maxrantil** [comment]
Thanks for the input, @t-bast , I’m a Summer of Bitcoin intern, and this is my first implementation here. The project proposal, authored by @EndotermicDev , originally included a BOLT proposal as part of the plan. But honestly, I didn’t know about the bLIP process until you brought it up.

That said, I believe that enhancing privacy is generally useful, so I figured it was worth adding. However, I’m totally open to hearing where others think this should go—whether it’s a BOLT or a bLIP.

File: 02-peer-protocol.md [hunk 1]
```diff
@@ -2377,6 +2378,52 @@ be trimmed at the updated feerate, this could overflow the configured
 `max_dust_htlc_exposure_msat`. Whether to close the channel preemptively
 or not is left as a matter of node policy.
 
+### Alternative Addresses: `alt_addr`
```
Comment (yyforyongyu) [inline 02-peer-protocol.md:2381]: Think this should be done in either a BOLT01 `init` msg or a BOLT07 `node_announcement` msg, and not in the channel state machine.

**Review by yyforyongyu — COMMENTED** [review]
Agree if we wanna include this, it's better to be put in a bLIP.

**Comment by t-bast** [comment]
> That said, I believe that enhancing privacy is generally useful

It's really unclear to me how this _generally_ enhances privacy. It rather looks like it enhances privacy only for some very specific use-cases and nodes, and isn't a general mechanism that we'd expect all nodes to use. That's why it's rather a candidate for a bLIP than the BOLTs, which we don't want to clutter with too many optional features.

**Comment by maxrantil** [comment]
this is now moved to bLIP 43 [https://github.com/lightning/blips/pull/43](url) . Thanks everyone for your input! Closing this out.

**Pull request closed without merging**